src/NuGet.Services.Metadata.Catalog.Monitoring/Status/PackageMonitoringStatusService.cs

```diff
@@ -66,20 +67,34 @@ public Task<PackageMonitoringStatus> GetAsync(FeedPackageIdentity package, Cance
             var packageStatuses = new List<PackageMonitoringStatus>();
 
             var storage = GetStorage(state);
-            foreach (var listItem in await storage.List(token))
-            {
-                var packageStatus = await GetPackageAsync(storage, listItem.Uri, token);
-
-                if (packageStatus == null)
+            return (await Task.WhenAll((await storage.List(token)).Select(listItem =>
+                Task.Run(async () =>
                 {
```

Comment:
This chaining of tasks is hard to read... almost looks like the Task.WhenAll is for the already awaited list tasks. I'd break it up and save the tasks to a local var and run WhenAll on a subsequent line.